crates/csv2arrow/src/main.rs

```diff
@@ -66,7 +68,12 @@ fn main() -> Result<(), ArrowError> {
 
     let mut file = File::open(&opts.input)?;
 
-    let mut input: Box<dyn SeekRead> = if file.rewind().is_ok() {
+    let mut input: Box<dyn SeekRead> = if opts.input.extension() == Some(OsStr::new("gz")) {
```

Comment:
Is that the safest way to check whether a file is compressed?

Reply:
I think if the extension is gz, we are good. I was just wondering whether people often have gzipped files that don't use the file extension. If the answer is that usually the extension is a good heuristic, I think what you have right now is great.

Reply:
gzip produces .gz, so I think checking for magic number is an overkill. I'd just skip this checking, but it's your decision.